ts/client/src/accounts/mangoAccount.ts

```diff
@@ -607,6 +613,33 @@ export class MangoAccount {
     return toUiDecimals(maxWithdrawWithBorrow, group.getMintDecimals(mintPk));
   }
 
+  public calculateEquivalentSourceAmount(
+    sourceBank: Bank,
+    targetBank: Bank,
+    targetRemainingDepositLimit: BN,
+  ): BN {
+    const decimalDifference = sourceBank.mintDecimals - targetBank.mintDecimals;
+    let adjustedExchangeRate: number;
+
+    // Adjust the exchange rate based on the decimal difference
+    if (decimalDifference > 0) {
```

Comment:
IMO all this is not needed, you can simply replace this whole method by `targetRemainingDepositLimit .mul(targetBank.price.div(sourceBank.price))`, this should take care of native native conversion, also you shouldnt need to do the if else